src/libraries/Microsoft.PowerFx.Core/IR/StackMarker.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.IR
+{
+    internal struct StackMarker
+    {
+        private readonly int _depth;
+        private readonly int _maxCallDepth;
+
+        internal StackMarker Inc()
+        {
+            if (_depth >= _maxCallDepth)
+            {
+                throw new MaxCallDepthException();
+            }
+
+            return new StackMarker(_depth + 1, _maxCallDepth);
+        }
+
+        public StackMarker(int depth, int maxCallDepth)
```

Comment:
Is there anywhere where depth is not 0? We might want to rename depth to currentDepth. We could swap these parameters, then make currentDepth = 0 a default value.

Reply:
Changed to this.